uaclient/cli.py

```diff
@@ -1216,7 +1216,10 @@ def action_detach(args, *, cfg) -> int:
 
     @return: 0 on success, 1 otherwise
     """
-    return _detach(cfg, assume_yes=args.assume_yes)
+    ret = _detach(cfg, assume_yes=args.assume_yes)
```

Comment:
I think we should only start the daemon if `ret` is `0` right ?

Reply:
ah good catch